packages/adapter-knex/lib/adapter-knex.js

```diff
@@ -507,6 +511,13 @@ class QueryBuilder {
 
     this._addWheres(w => this._query.andWhere(w), listAdapter, where, baseTableAlias);
 
+    // TODO: Implement configurable search fields for lists
+    const searchField = listAdapter.fieldAdaptersByPath['name'];
+    if (search !== undefined && searchField && searchField.fieldName === 'Text') {
+      this._query.whereRaw('true');
+      this._query.andWhere(`${baseTableAlias}.name`, `LIKE`, `%${search}%`);
```

Comment:
We need to apply `escapeRegExp` to `search` to ensure special characters in the search don't get interpreted as regex controls.